magma/conversions.py

```diff
@@ -195,6 +196,8 @@ def concat(*arrays):
             ts.extend(a.bits())
         elif isinstance(a, Bit):
             ts.extend([a])
+        elif isinstance(a, (bool, ht.Bit)):
+            ts.extend([Bit(a)])
         else:
```

Comment:
also just noting it because I see it now, not sure how robust this else is 😬

Reply:
I think it assumes that `a: Array` and the above cases handle the implicit coercion to an array type.  We can add a type error if it's not an array.